docs/sources/tempo/operations/backend_search.md

```diff
@@ -141,7 +203,196 @@ querier:
+In a shared cluster, you need to use more caution when increasing `querier.max_outstanding_per_tenant`
+
+### Parameter interactions
+
+The `query_frontend` configuration options control how Tempo shards a query into multiple jobs.
+These options also control the size of a single job, number of jobs, and the number of jobs added to the work queue at once (concurrently) for queriers to be picked up.
+The `max_batch_size` option controls how many jobs are sent to queriers at once in a single batch.
+
+Querier configuration options control processing of these jobs, queries operate at a job level.
+They're workers: they pick a job from the queue and execute the job, and return the results back to the query-frontend.
+Querier options control how many jobs it processes at once by opening concurrent connections to query-frontends, and each connection processes a single.
+
+## Guidelines on key configuration parameters
+
+The following sections provide recommendations for key configuration parameters.
+
+### `query_frontend.max_outstanding_per_tenant` parameter
+
+The `query_frontend.max_outstanding_per_tenant` parameter decides the maximum number of outstanding requests per tenant per frontend; requests beyond this error with HTTP `429`.
+
+This configuration controls how much load a single tenant can put on a cluster. A return of `429` indicates users to slow down.
+
+This configuration can be used to maintain quality of service (QoS) for tenants by asking tenants to slow down by sending `429` to the tenant that's overloading the system.
+
+#### Guidelines
+
+* In a dedicated cluster with one big tenant, it’s okay to increase the number for `query_frontend.max_outstanding_per_tenant.`
+* In a shared cluster with a lot of small tenants, keep the number small.
+* If a single tenant is overwhelming the whole cluster, you should decrease this parameter. It reduces the amount of work the tenant can enqueue at once. It starts returning `429` to the tenant.
+* If your tenants are complaining about getting `429`, you might want to increase this parameter and scale out queries to handle the query load.
+* If you increase other configurations that increase the query throughput or scale queriers, you might want to reduce this parameter to control the amount of work that can be enqueued.
+
+
+### `query_frontend.max_batch_size` parameter
+
+The number of jobs to batch together in one HTTP request to the querier.
+Large search queries over a longer time range produce thousands of jobs.
+Instead of sending jobs one by one, batch the jobs and send them at once.
+The querier picks up and process them.
+
+Batching pushes jobs faster to the queriers, and reduce the time spent waiting around for jobs to be picked up by the querier.
+
+#### Guidelines
+
+* Default value of `max_batch_size` is set to `5`. In testing, `5` is a good default across all sizes of clusters.
```

Comment:
```suggestion
* Default value of `max_batch_size` is set to `5`. In our testing we found that `5` is a good default across all sizes of clusters.
```

Reply:
I think we could remove the second sentence here entirely - "In our testing, we found that..." isn't necessary because the next buillet says DO NOT change the size.